runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/retries/delay/StandardRetryTokenBucket.kt

```diff
@@ -100,48 +112,127 @@ public class StandardRetryTokenBucket constructor(
-            timeoutRetryCost = 10,
-            initialTryCost = 0,
-            initialTrySuccessIncrement = 1,
-        )
+        public val initialTrySuccessIncrement: Int = builder.initialTrySuccessIncrement
+
+        /**
+         * The maximum capacity for the bucket
+         */
+        public val maxCapacity: Int = builder.maxCapacity
+
+        /**
+         * The amount of capacity to return per second. Setting this to 0 automatically sets [circuitBreakerMode] to
+         * `true`.
+         */
+        public val refillUnitsPerSecond: Int = builder.refillUnitsPerSecond
+
+        /**
+         * The amount of capacity to decrement for standard retries
+         */
+        public val retryCost: Int = builder.retryCost
+
+        /**
+         * The amount of capacity to decrement for timeout or throttling retries
+         */
+        public val timeoutRetryCost: Int = builder.timeoutRetryCost
+
+        @InternalApi
+        override fun toBuilderApplicator(): RetryTokenBucket.Config.Builder.() -> Unit = {
+            if (this is Builder) {
+                circuitBreakerMode = this@Config.circuitBreakerMode
+                initialTryCost = this@Config.initialTryCost
+                initialTrySuccessIncrement = this@Config.initialTrySuccessIncrement
+                maxCapacity = this@Config.maxCapacity
+                refillUnitsPerSecond = this@Config.refillUnitsPerSecond
+                retryCost = this@Config.retryCost
+                timeoutRetryCost = this@Config.timeoutRetryCost
+            }
+        }
+
+        /**
+         * A mutable builder for a [Config]
+         */
+        public class Builder : RetryTokenBucket.Config.Builder {
+            /**
+             * When `true`, indicates that attempts to acquire tokens or schedule retries should fail if all capacity
+             * has been depleted. When `false`, calls to acquire tokens or schedule retries will delay until sufficient
+             * capacity is available. This property will automatically be set to `true` if [refillUnitsPerSecond] is 0.
+             */
+            public var circuitBreakerMode: Boolean = true
+
+            /**
+             * The amount of capacity to decrement for the initial try
+             */
+            public var initialTryCost: Int = 0
+
+            /**
+             * The amount of capacity to return if the initial try is successful
+             */
+            public var initialTrySuccessIncrement: Int = 1
+
+            /**
+             * The maximum capacity for the bucket
+             */
+            public var maxCapacity: Int = 500
+
+            /**
+             * The amount of capacity to return per second. Setting this to 0 automatically sets [circuitBreakerMode] to
+             * `true`.
+             */
+            public var refillUnitsPerSecond: Int = 0
+                set(value) {
+                    if (value == 0) circuitBreakerMode = true
+                    field = value
```

Comment:
No, circuit-breaker mode is not required when the refill rate is non-zero. That mode just changes what happens when callers request retry capacity while the tokens are exhausted: throw an exception (`circuitBreakerMode = true`) or delay (`circuitBreakerMode = false`). A refill rate of zero means that only tries will return capacity and thus we shouldn't circuit break (otherwise, retry calls would hang indefinitely).